packages/vite/src/node/server/index.ts

```diff
@@ -726,12 +740,15 @@ async function startServer(
   const port = inlinePort ?? options.port ?? DEFAULT_DEV_PORT
   const hostname = await resolveHostname(options.host)
 
-  await httpServerStart(httpServer, {
+  _configServerPort = port
```

Comment:
Could we do the check to preserve the current port here instead of on `_createServer`, so we keep the logic contained. I think this should work, no?
```js
  const options = server.config.server
  const hostname = await resolveHostname(options.host)

  const configPort = inlinePort ?? options.port
  const port = ((!configPort || configPort === _configServerPort) ? _currentServerPort : configPort ) ?? DEFAULT_DEV_PORT
  _configServerPort = configPort
```

Reply:
As you said, it seems that we can modify it here. However, the logic for determining whether to print out the URL seems to need to be modified.
https://github.com/vitejs/vite/blob/ac2503d52cfc071b5be07485183708dee23b82c5/packages/vite/src/node/server/index.ts#L960

```ts
  (_currentServerPort ?? port ?? DEFAULT_DEV_PORT) !== (prevPort ?? DEFAULT_DEV_PORT)
```